styleguide/js/zoolander.js

```diff
@@ -126,6 +128,114 @@ jQuery(document).ready(function($) {
   };
 }(jQuery)); //to protect and scope the JQuery alias = $
 
+// Filter bar plugin
+(function($){
+  'use strict';
+  $.fn.rsFilterBar = function(){
+    // Variable declorations
+    var $filterBar = $(this);
+    var $filterForm = $filterBar.find('.rsFilter-form');
+    var $stickyItemTop = $filterForm.offset().top;
+    var $mainNavHeight = parseInt($('.navbar-fixed-top').outerHeight(), 10);
+    var $barPadding = parseInt($filterBar.css('padding-left'), 10) +
+                      parseInt($filterBar.css('padding-right'), 10);
+    var $filterHamburger = $filterBar.find('.rsFilter-hamburger');
+    var $rsWindow = $(window);
+    var isFixed = false;
+    // var inBounds = false;
+
+    // Callback to get new form width on resizing/scrolling
+    function getWidth(bar, padding) {
+      var width = parseInt(bar.outerWidth(true), 10) - padding;
+      return width;
+    }
+
+    // Callback to check if in bounds of side menu
+    function inBounds(wt) {
+      return $stickyItemTop - $mainNavHeight < wt &&
+        $filterBar.outerHeight() + $filterBar.offset().top +
+        $mainNavHeight - ($filterForm.outerHeight() * 3) >
+        (wt + $filterForm.outerHeight())
+    }
+
+    $filterHamburger.click(function() {
+      $filterForm.slideToggle(150);
+    });
+
+    $rsWindow.scroll(function() {
+      // If not in mobile
+      if ($rsWindow.outerWidth() > 767) {
+        var $rsWindowTop = $rsWindow.scrollTop();
+        // If user scrolls to filter form and is in the bounds of the bar, move form up and down
+        if (inBounds($rsWindowTop)) {
+          $filterForm.css({
+            'position' : 'fixed',
+            'top' : $mainNavHeight + 10,
+            'width' : getWidth($filterBar, $barPadding) + 'px'
+          });
+          isFixed = true;
+        } else {
+          $filterForm.css({
+            'position' : '',
+            'width' : ''
+          });
+          isFixed = false;
+        }
+      }
+    });
+
+    // Set a slight delay to the resize functionality so that it doesn't
+    // run on every pixel of a browser resize
+    var resizeTimer;
+    $rsWindow.resize(function() {
+      clearTimeout(resizeTimer);
+      resizeTimer = setTimeout(function() {
+      var $rsWindowTop = $rsWindow.scrollTop();
+        // If not mobile
+        if ( $rsWindow.outerWidth() > 768 ) {
+          if (!isFixed && inBounds($rsWindowTop)) {
+            $filterForm.css({
+              'position' : 'fixed',
+              'width' : getWidth($filterBar, $barPadding) + 'px'
+            });
+            isFixed = true;
+          } else if (isFixed && inBounds($rsWindowTop)) {
+            $filterForm.css({
+              'width' : getWidth($filterBar, $barPadding) + 'px'
+            });
+          } else if (isFixed && !inBounds($rsWindowTop)) {
```

Comment:
also it can't just be stored once, you have to keep checking if you are in bounds of the side bar when you're scrolling up and down the page. It's not always going to be the same value.